gravitee-apim-console-webui/src/management/api/notifications/notifications-list/notofication-details/notification-details.component.ts

```diff
@@ -13,11 +13,107 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component } from '@angular/core';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { takeUntil, tap } from 'rxjs/operators';
+import { combineLatest, Subject } from 'rxjs';
+import { groupBy, keysIn } from 'lodash';
+import { FormControl, FormGroup } from '@angular/forms';
+
+import { UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { NotificationSettingsService } from '../../../../../services-ngx/notification-settings.service';
+import { SnackBarService } from '../../../../../services-ngx/snack-bar.service';
+import { NotificationSettings } from '../../../../../entities/notification/notificationSettings';
+import { Notifier } from '../../../../../entities/notification/notifier';
+import { Hooks } from '../../../../../entities/notification/hooks';
+
+type HooksForm = {
+  list: any;
```

Comment:
What do you think about also avoiding this any ? 

I wonder if it wouldn't be better to have a structure like this? 

```ts
type CategoriesHooksVM = {
  categoryName: string; 
  hooks: Hooks & {checked: boolean};
}[]
```
as after you use it to make loop inside loop

Reply:
I changed a little bit the way I'm building this object. The issue was that I reorganized object structure in last line of ngOnInit, maybe this one is better.

Reply:
Done